File: csharp/DVSClient/DVSClient/phone/validate/Result.cs
using DVSClient.Common;
using DVSClient.Server.Phone;

namespace DVSClient.Phone.Validate
{
    public class Result
    {
        public string Number { get; }
        public string ValidatedPhoneNumber { get; }
        public string FormattedPhoneNumber { get; }
        public PhoneType? PhoneType { get; }
        public Confidence? Confidence { get; }
        public string PortedDate { get; }
        public string DisposableNumber { get; }
        public Metadata? Metadata { get; }

        public Result(RestApiPhoneValidateResponse response)
        {
            var result = response.Result;
            if (result != null)
            {
                Number = result.Number ?? string.Empty;
                ValidatedPhoneNumber = result.ValidatedPhoneNumber ?? string.Empty;
                FormattedPhoneNumber = result.FormattedPhoneNumber ?? string.Empty;
                PhoneType = result.PhoneType?.GetEnumValueFromJsonName<PhoneType>();
                Confidence = result.Confidence?.GetEnumValueFromJsonName<Confidence>();
                PortedDate = result.PortedDate ?? string.Empty;
                DisposableNumber = result.DisposableNumber ?? string.Empty;
            }
            else
            {
                Number = string.Empty;
                ValidatedPhoneNumber = string.Empty;
                FormattedPhoneNumber = string.Empty;
                PhoneType = null;
                Confidence = null;
                PortedDate = string.Empty;
                DisposableNumber = string.Empty;
            }

            var metadataFromApi = response.Metadata;
            Metadata = metadataFromApi != null ? new Metadata(metadataFromApi) : null;
        }
    }
}

File: java/src/main/java/com/experian/dvs/client/phone/validate/Result.java
package com.experian.dvs.client.phone.validate;

import com.experian.dvs.client.phone.Confidence;
import com.experian.dvs.client.server.phone.RestApiPhoneValidateMetadata;
import com.experian.dvs.client.server.phone.RestApiPhoneValidateResponse;
import com.experian.dvs.client.server.phone.RestApiPhoneValidateResult;

import java.util.Objects;

public class Result {

    private final String number;
    private final String validatedPhoneNumber;
    private final String formattedPhoneNumber;
    private final String phoneType;
    private final Confidence confidence;
    private final String portedDate;
    private final String disposableNumber;
    private final Metadata metadata;


    public Result(final RestApiPhoneValidateResponse response) {

        final RestApiPhoneValidateResult result = response.getResult();
        if (result != null) {
            this.number = Objects.toString(result.getNumber(), "");
            this.validatedPhoneNumber = Objects.toString(result.getValidatedPhoneNumber(), "");
            this.formattedPhoneNumber = Objects.toString(result.getFormattedPhoneNumber(), "");
            this.phoneType = Objects.toString(result.getPhoneType(), "");
            this.confidence = result.getConfidence() != null ? Confidence.fromValue(result.getConfidence()): Confidence.UNKNOWN;
            this.portedDate = Objects.toString(result.getPortedDate(), "");
            this.disposableNumber = Objects.toString(result.getDisposableNumber(), "");
        } else {
            this.number = "";
            this.validatedPhoneNumber = "";
            this.formattedPhoneNumber = "";
            this.phoneType = "";
            this.confidence = Confidence.UNKNOWN;
            this.portedDate = "";
            this.disposableNumber = "";
        }

        final RestApiPhoneValidateMetadata metadataFromApi = response.getMetadata();
        if (metadataFromApi != null) {
            this.metadata = new Metadata(metadataFromApi);
        } else {
            this.metadata = null;
        }

    }

    public String getNumber() {
        return number;
    }

    public String getValidatedPhoneNumber() {
        return validatedPhoneNumber;
    }

    public String getFormattedPhoneNumber() {
        return formattedPhoneNumber;
    }

    public String getPhoneType() {
        return phoneType;
    }

    public Confidence getConfidence() {
        return confidence;
    }

    public String getPortedDate() {
        return portedDate;
    }

    public String getDisposableNumber() {
        return disposableNumber;
    }

    public Metadata getMetadata() {
        return metadata;
    }
}
